samples/cpp/digits.cpp

```diff
@@ -1,375 +1,185 @@
-
-    for (size_t i = 0; i < digits.size(); i++)
-    {
-        shuffled_indexes[i] = (int)i;
-    }
-
-    randShuffle(shuffled_indexes);
-
-    vector<Mat> shuffled_digits(digits.size());
-    vector<int> shuffled_labels(labels.size());
-
-    for (size_t i = 0; i < shuffled_indexes.size(); i++)
-    {
-        shuffled_digits[shuffled_indexes[i]] = digits[i];
-        shuffled_labels[shuffled_indexes[i]] = labels[i];
-    }
-
-    digits = shuffled_digits;
-    labels = shuffled_labels;
+    Mat probMat = probBlob.reshape(1, 1);
+    Point classNumber;
+    minMaxLoc(probMat, NULL, classProb, NULL, &classNumber);
+    *classId = classNumber.x;
 }
 
-int main(int /* argc */, char* argv[])
+void predictor(dnn::Net net, Mat &roi, int &classId, double &probability)
 {
-    help(argv);
-
-    vector<Mat> digits;
-    vector<int> labels;
-
-    load_digits(DIGITS_FN, digits, labels);
-
-    cout << "preprocessing..." << endl;
-
-    // shuffle digits
-    shuffle(digits, labels);
-
-    vector<Mat> digits2;
-
-    for (size_t i = 0; i < digits.size(); i++)
-    {
-        Mat deskewed_digit;
-        deskew(digits[i], deskewed_digit);
-        digits2.push_back(deskewed_digit);
-    }
-
-    Mat samples;
-
-    preprocess_hog(digits2, samples);
-
-    int train_n = (int)(0.9 * samples.rows);
-    Mat test_set;
-
-    vector<Mat> digits_test(digits2.begin() + train_n, digits2.end());
-    mosaic(25, digits_test, test_set);
-    imshow("test set", test_set);
-
-    Mat samples_train = samples(Rect(0, 0, samples.cols, train_n));
-    Mat samples_test = samples(Rect(0, train_n, samples.cols, samples.rows - train_n));
-    vector<int> labels_train(labels.begin(), labels.begin() + train_n);
-    vector<int> labels_test(labels.begin() + train_n, labels.end());
-
-    Ptr<ml::KNearest> k_nearest;
-    Ptr<ml::SVM> svm;
-    vector<float> predictions;
-    Mat vis;
-
-    cout << "training KNearest..." << endl;
-    k_nearest = ml::KNearest::create();
-    k_nearest->train(samples_train, ml::ROW_SAMPLE, labels_train);
-
-    // predict digits with KNearest
-    k_nearest->findNearest(samples_test, 4, predictions);
-    evaluate_model(predictions, digits_test, labels_test, vis);
-    imshow("KNearest test", vis);
-    k_nearest.release();
-
-    cout << "training SVM..." << endl;
-    svm = ml::SVM::create();
-    svm->setGamma(5.383);
-    svm->setC(2.67);
-    svm->setKernel(ml::SVM::RBF);
-    svm->setType(ml::SVM::C_SVC);
-    svm->train(samples_train, ml::ROW_SAMPLE, labels_train);
-
-    // predict digits with SVM
-    svm->predict(samples_test, predictions);
-    evaluate_model(predictions, digits_test, labels_test, vis);
-    imshow("SVM test", vis);
-    cout << "Saving SVM as \"digits_svm.yml\"..." << endl;
-    svm->save("digits_svm.yml");
-    svm.release();
-
-    waitKey();
-
-    return 0;
+    Mat pred;
+    //Convert Mat to batch of images
+    Mat inputBlob = dnn::blobFromImage(roi, 1, Size(28, 28), Scalar(), false);
```

Comment:
Please don't use default arguments:
```
    Mat inputBlob = dnn::blobFromImage(roi, 1, Size(28, 28));
```

Reply:
Thanks for your code review, I am not sure that my newly submitted code is modified correctly.